src/gui/lineedit/src/lineedit.cpp

```diff
@@ -8,37 +8,34 @@
 ****************************************************************************/
 
 #include "lineedit.h"
+
 #include <algorithm>
+
+#include <QResizeEvent>
 #include <QStyle>
 #include <QToolButton>
-#include <QResizeEvent>
+
+#include "guiiconprovider.h"
 
 LineEdit::LineEdit(QWidget *parent)
     : QLineEdit(parent)
 {
-    QPixmap pixmap1(":/lineeditimages/search.png");
-    searchButton = new QToolButton(this);
-    searchButton->setIcon(QIcon(pixmap1));
-    searchButton->setIconSize(pixmap1.size());
-    searchButton->setCursor(Qt::ArrowCursor);
-    searchButton->setStyleSheet("QToolButton { border: none; padding: 2px; }");
-    QSize searchButtonHint = searchButton->sizeHint();
-
-    QSize clearButtonHint(0, 0);
+    m_searchButton = new QToolButton(this);
+    m_searchButton->setIcon(GuiIconProvider::instance()->getIcon("edit-find"));
+    m_searchButton->setCursor(Qt::ArrowCursor);
+    m_searchButton->setStyleSheet("QToolButton { border: none; padding: 2px; }");
+
+    setStyleSheet(QString("QLineEdit { padding-left: %1px; }").arg(m_searchButton->sizeHint().width()));  // padding between text and widget borders
```

Comment:
This line is too long. At least, move the comment to another line.

Reply:
Then I can see it's not your code... so I don't insist.

Reply:
Moved the comment to above